src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/model/person/name/Names.kt

```diff
@@ -0,0 +1,40 @@
+package uk.gov.justice.digital.hmpps.personrecord.model.person.name
+
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.NameEntity
+import uk.gov.justice.digital.hmpps.personrecord.model.types.NameType
+
+data class Names(
+  val preferred: Name? = null,
+  val aliases: List<Name> = emptyList(),
+  val nickname: Name? = null,
+  val maiden: Name? = null,
+) {
+  fun build(): MutableList<NameEntity> {
+    val names: List<Name> = listOfNotNull(preferred, nickname, maiden) + aliases
+    return NameEntity.fromList(names).toMutableList()
+  }
```

Comment:
🤔 
This might be the best option but it does rather look as if it belongs on another class ,maybe one which does not exist? A wrapper for a collection of `NameEntity` perhaps? Not sure it would be worth the JPA hassle